File: src/courses/data/catalog/02-dsa/dsa-core/implementations/csharp/searching.cs
/**
 * C# Implementation: Searching (Linear & Binary)
 * Invariant: Linear O(N), Binary O(log N)
 */
using System;
using System.Collections.Generic;

namespace Seed.DSA.Core
{
    public class SearchingLinearBinary<T>
    {
        private readonly List<T> _items = new List<T>();

        // TODO: Implement algorithm operations
        public void Add(T item)
        {
            _items.Add(item);
        }

        public int Count => _items.Count;
    }
}

File: src/courses/data/catalog/02-dsa/dsa-core/implementations/java/searching.java
/**
 * Java Implementation: Searching (Linear & Binary)
 * Invariant: Linear O(N), Binary O(log N)
 */
package seed.dsa.core;

import java.util.*;

public class SearchingLinearBinary<T> {
    private final List<T> items = new ArrayList<>();

    // TODO: Implement your solution here
    public void add(T value) {
        items.add(value);
    }

    public int size() {
        return items.size();
    }
}
